java/core/src/main/java/com/lancedb/lance/Dataset.java

```diff
@@ -210,6 +259,27 @@ public long latestVersion() {
 
   private native long nativeLatestVersion();
 
+  /**
+   * Creates a new index on the dataset.
+   * Only vector indexes are supported.
+   *
+   * @param columns   the columns to index from
+   * @param indexType the index type
+   * @param name      the name of the created index
+   * @param params    index params
+   * @param replace   whether to replace the existing index
+   */
+  public void createIndex(List<String> columns, IndexType indexType, Optional<String> name,
+      IndexParams params, boolean replace) {
+    try (LockManager.WriteLock writeLock = lockManager.acquireWriteLock()) {
```

Comment:
why do we need a lock here?

Reply:
This is a lock to ensure the Dataset is not closed and the Dataset corresponding rust JNI object is still valid
Good catch, all operations is read lock which do not prevent each other